### ISSUE TITLE
Protocol member not verified at class definition with explicit subclasses

### ISSUE THREAD

**Issue by BaconPancakes** [microsoft/pyright#7545]
**Describe the bug**
According to https://typing.readthedocs.io/en/latest/spec/protocol.html#explicitly-declaring-implementation
when a `Protocol` is explicitly subclassed, the type checker can verify the class has implemented the Protocol correctly.

This used to be available in `pyright`, and it was a very useful feature. Not sure if this was purposely removed or somehow disabled from defaults.

**Code or Screenshots**
Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYCwAULaRQIYDOzUAsnAJIowCmIYI1J8AFETAlyFAJQAuWlCVQIcAPqMYMEEgBGAV35yozbbUXKAJn2Ar1wfSlHM%2BFYDKgBaAHyZeCmmUgqAA6MPMaBhY2Ti5sCj4IPl5NJDAnWN4BIRF5CyUUdJcUFw0tHQN%2BKABeKABWfKgAYig%2BAA8EPlJ%2BS1hEPlaQcBA2YFwVJFZUDFUy7T1DAcYUXtmHJxlGlsqoQpQBgHcBAZA%2BHDw%2BS1ogA)

```python
from typing import Protocol

class MyInterface(Protocol):
    my_attribute: str

    def my_fun(self) -> int:
        ...

class MyImplementation(MyInterface):
    nonsense_attribute = 5
    # expected type errors for missing my_attribute and my_fun()
    # but none were reported

```

**VS Code extension or command-line**
Pylance

**Comment by erictraut** [comment]
Pyright is working as intended here and is compliant with the typing spec. I don't consider this a bug.

If you define an unimplemented method in the protocol (as in your example with `my_fun`), the typing spec indicates that if an explicit non-protocol subclass does not implement this method, the subclass should be treated as an abstract base class. Unless this class is marked `@final` or an attempt is made to instantiate the (abstract) class, it is not considered an error. You'll note that if you add an `@final` decorator to `MyImplementation`, pyright will generate an error. Likewise, if you attempt to instantiate `MyImplementation`, pyright will generate an error.

For an attribute, there's no definition in the typing spec what it means to be "implemented". Type checkers do not generally verify that a declared attribute has been assigned a value.
